### ISSUE TITLE
Document how to use filters

### ISSUE THREAD

**Issue by khaledhosny** [googlei18n/ufo2ft#240]
I’m trying to use `PropagateAnchorsFilter` in my UFOs, but I couldn’t find any documentation and had to navigate through the source code (and still navigating).

**Comment by anthrotype** [comment]
Thanks. Yes, sorry, documentation is none at the moment. 

I suggest you have a look at the `ufo2ft/filters/__init__.py`, especially `the loadFilters` and `getFilterClass` functions.

the lib key is "com.github.googlei18n.ufo2ft.filters", contains a plist array of dicts, each dict has a required "name" string (e.g. "Propagate Anchors"), then optional "include" or "exclude" array of strings (glyph names), and depending on the filter class, required positional "args" (plist array) and/or optional "kwargs" (plist dict); finally a "pre" (bool) element to determine whether a filter should happen before (True) or after (False) the default filters.

See the [ufo2ft.preProcessor](https://github.com/googlei18n/ufo2ft/blob/master/Lib/ufo2ft/preProcessor.py) module which manages loading and running the default and custom filters.

**Comment by khaledhosny** [comment]
Thanks, that is pretty much what I found at the end (then discover that the latest does not have that filter yet :smiley:).

**Comment by anthrotype** [comment]
also see https://github.com/googlei18n/glyphsLib/blob/1c9a32266017312e247a432a78e7ed5d85e0c850/tests/builder/builder_test.py#L276-L291

**Comment by anthrotype** [comment]
I think we can cut a release tomorrow. /cc @moyogo

**Comment by khaledhosny** [comment]
I’m testing with master now, but I’m not getting my anchors propagated, investigating.

**Comment by khaledhosny** [comment]
Hmm, looks like fontmake decomposes components before calling ufo2ft compiler, so it is too late for the filter and no anchors gets propagated.

**Comment by anthrotype** [comment]
Yep. I have a wip branch that makes fontmake use ufo2ft filters for decomposing, removing overlaps and cubic2quadratic conversion. i’ll push soon.
